client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
      * 
      * @param stream Stream object.
      * @param sealedSegments List of segments to be sealed.
      * @param newKeyRanges Key ranges after scaling the stream.
+     * @param timeout timeout and throw exception if scale does not complete withing specified timeout.
+     * @param executorService executor to be used for busy waiting.
      * @return A future which will throw if the operation fails, otherwise returning a boolean to
-     *         indicate that the stream was scaled or false if the segments to seal have already
-     *         been sealed.
+     *         indicate that the scaling was started or not.
      */
     CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
```

Comment:
This is on the controller interface.. controller interface given out to client doesnt have to map 1:1 to underlying rpc calls.
I feel this is exactly the place where this belongs. Please let me know if you think otherwise.